diff --git a/scripts/bin2coff.c b/scripts/bin2coff.c
index f9d30eeb0..d9bfae32f 100644
--- a/scripts/bin2coff.c
+++ b/scripts/bin2coff.c
@@ -61,6 +61,7 @@ typedef unsigned long long   uint64_t;
 #define IMAGE_FILE_MACHINE_I386			 0x014c
 #define IMAGE_FILE_MACHINE_IA64			 0x0200
 #define IMAGE_FILE_MACHINE_AMD64		 0x8664
+#define IMAGE_FILE_MACHINE_ARM64		 0xAA64
 
 #define IMAGE_FILE_RELOCS_STRIPPED		 0x0001
 #define IMAGE_FILE_EXECUTABLE_IMAGE		 0x0002
@@ -231,10 +232,16 @@ typedef struct {
 
 #pragma pack(pop)
 
+static int arm64 = 0;
+
 static int check_64bit(const char *arg, int *x86_32)
 {
 	if ((strcmp(arg, "64bit") == 0) || (strcmp(arg, "x64") == 0))
 		*x86_32 = 0; /* 0 = 64bit */
+	else if ((strcmp(arg, "ARM64") == 0) || (strcmp(arg, "arm64") == 0)) {
+		*x86_32 = 0; /* same symbol naming as x64 */
+		arm64 = 1;
+	}
 	else if ((strcmp(arg, "32bit") == 0) || (strcmp(arg, "Win32") == 0))
 		*x86_32 = 1; /* 1 = 32bit */
 	else
@@ -262,7 +269,7 @@ main (int argc, char *argv[])
 	SIZE_TYPE* data_size;
 
 	if ((argc < 3) || (argc > 5)) {
-		fprintf(stderr, "\nUsage: bin2coff bin obj [label] [64bit|Win32|x64]\n\n");
+		fprintf(stderr, "\nUsage: bin2coff bin obj [label] [64bit|Win32|x64|ARM64]\n\n");
 		fprintf(stderr, "  bin  : source binary data\n");
 		fprintf(stderr, "  obj  : target object file, in MS COFF format.\n");
 		fprintf(stderr, "  label: identifier for the extern data. If not provided, the name of the\n");
@@ -341,7 +348,7 @@ main (int argc, char *argv[])
 	string_table = (IMAGE_STRINGS*)&buffer[sizeof(IMAGE_FILE_HEADER) + sizeof(IMAGE_SECTION_HEADER) + size + sizeof(SIZE_TYPE) + 2*sizeof(IMAGE_SYMBOL)];
 
 	/* Populate file header */
-	file_header->Machine = (x86_32)?IMAGE_FILE_MACHINE_I386:IMAGE_FILE_MACHINE_AMD64;
+	file_header->Machine = (x86_32)?IMAGE_FILE_MACHINE_I386:(arm64?IMAGE_FILE_MACHINE_ARM64:IMAGE_FILE_MACHINE_AMD64);
 	file_header->NumberOfSections = 1;
 	file_header->PointerToSymbolTable = sizeof(IMAGE_FILE_HEADER) + sizeof(IMAGE_SECTION_HEADER) + (uint32_t)size+4;
 	file_header->NumberOfSymbols = 2;
diff --git a/scripts/tesseract/endianness.h b/scripts/tesseract/endianness.h
index eec9e74c1..e7bf4325a 100644
--- a/scripts/tesseract/endianness.h
+++ b/scripts/tesseract/endianness.h
@@ -13,7 +13,7 @@
 	defined(__AARCH64EL__) ||					\
 	defined(_MIPSEL) || defined(__MIPSEL) || defined(__MIPSEL__) ||	\
 	defined(_M_IX86) || defined(_M_X64) ||				\
-	defined(_M_IS64) || defined(_M_ARM)
+	defined(_M_IS64) || defined(_M_ARM) || defined(_M_ARM64)
 // It's a little-endian target architecture
 #   define L_LITTLE_ENDIAN
 #else
